marlowe-playground-client/src/Simulator/State.purs

```diff
@@ -223,6 +191,20 @@ updatePossibleActions oldState@{ executionState: SimulationRunning executionStat
 
 updatePossibleActions oldState = oldState
 
+extractRequiredActionsWithTxs :: TransactionInput -> State -> Term T.Contract -> Tuple State (Array Action)
+extractRequiredActionsWithTxs txInput state contract
+  | TransactionOutput { txOutContract, txOutState } <- computeTransaction txInput state contract = Tuple txOutState (extractRequiredActions txOutContract)
+  | TransactionInput { inputs: Nil } <- txInput
+  , IntervalTrimmed env fixState <- fixInterval (unwrap txInput).interval state
+  , Just (_ /\ reducedContract) <- reduceContractUntilQuiescent env fixState contract = Tuple fixState (extractRequiredActions reducedContract)
+  -- the actions remain unchanged in error cases, cases where the contract is not reduced or cases where inputs remain
+  | otherwise = Tuple state (extractRequiredActions contract)
+
+extractRequiredActions :: Term T.Contract -> Array Action
+extractRequiredActions contract = case contract of
+  Term (When cases _ _) _ -> map (\(S.Case action _) -> action) $ mapMaybe fromTerm cases
+  _ -> mempty
+
```

Comment:
This was on the Marlowe.Semantic module but it had no implementation in haskell and it wasn't used outside the context of the simulator, so I moved it here. Eventually we could think of unifying this and the named actions in Marlowe.Dashboard.